### ISSUE TITLE
Enable configuration of relative panel width (resizable panels)

### ISSUE THREAD

**Issue by jfrer** [subugoe/tido#368]
In GitLab by @rmraschkowski on Nov 26, 2021, 12:13

Some panels are regarded less important than others by stakeholders/ users. E.g. Metadata panel doesn't need as much relative width as the Text panel, especially as the transcript tab needs to show the full lines without additional linebreaks to stay true to the manuscript. Being able to configure the relative width in a config file would be nice to have. E.g. "Manuscript = 25%; Text = 30%; Commentary = 25%; Metadata = 20%

**Comment by jfrer** [comment]
In GitLab by @rmraschkowski on Dec 8, 2021, 09:30

marked this issue as related to farbenlehre/gfl-app-new#103

**Comment by jfrer** [comment]
In GitLab by @mweidling on Jan 25, 2024, 08:48

unassigned @krvoigt

**Comment by jfrer** [comment]
In GitLab by @mweidling on Jan 25, 2024, 08:48

Ping. This is also relevant for the EUPT project.

**Comment by jfrer** [comment]
In GitLab by @paulpestov on Jan 25, 2024, 09:22

Could you try to set the panel width in your CSS? Since v3.0 we are exposing TIDO's CSS file, so your custom CSS can be inserted after that file to modify the styles. (No warranty for sanity of CSS)

**Comment by paulpestov** [comment]
After looking into this issue I'm not sure if percentages are the right choice to solve this. The problem is that we cannot say what is 100%. 

I would like to introduce a horizontal scalable panel arrangement. Right now the panels use the width from their parent container and split the widths equally. This already isn't the ideal solution if users want to add more than 4-5 panels. 

That's why I suggest to add a min width to a panel. On mobile devices we would maintain the horizonal approach. This would mean that a min width would span a almost the full width of the mobile screen and hint the adjacent panel visually like it's used be designed in mobiel card design, e.g. Play Store. 

Having that as basis I  then suggest to add a configuration key `width` at the panel config object. The value should not be a percentage but a ratio number. So the calculation goes like this: `[min width] * [value in config]`. 
An example setup with 4 panels could be: `1 1 2 1` which would show the 3rd panel twice as large as the rest. Values lower than 1 are not allowed. But we could expose the min width in pixels to the config, so it stays flexible.

@orlinmalkja @mweidling @tidoe @krvoigt

**Comment by mweidling** [comment]
> That's why I suggest to add a min width to a panel. On mobile devices we would maintain the horizonal approach. This would mean that a min width would span a almost the full width of the mobile screen and hint the adjacent panel visually like it's used be designed in mobiel card design, e.g. Play Store.

Good idea for smartphones. Would this behaviour be triggered for tablets as well? 

> An example setup with 4 panels could be: 1 1 2 1 which would show the 3rd panel twice as large as the rest. Values lower than 1 are not allowed. But we could expose the min width in pixels to the config, so it stays flexible.

Also +1 for this as long as floating point numbers are allowed (e.g. `1 1.5 2` for a three panel setup).

**Comment by tidoe** [comment]
Can you provide a way to turn off the horizontal overflow?
In EUPT, we had a panel that was twice as large as the other panels. We achieved this by setting `flex-grow: 2` for that panel. Now, this CSS command has no effect anymore. And when I try to use `width: 2` in the tido config, the panels don't fit into the window anymore (i.e. there is an unwanted horizontal overflow).

**Comment by paulpestov** [comment]
implemented in #572
